internal/k8s/reconciler/gateway.go

```diff
@@ -49,21 +49,22 @@ type K8sGatewayConfig struct {
 }
 
 func NewK8sGateway(gateway *gwv1beta1.Gateway, config K8sGatewayConfig) *K8sGateway {
+	// FUTURE (nathancoleman) See if we can avoid setting ConsulNamespace out of band
+	gwState := state.InitialGatewayState(gateway)
+	gwState.ConsulNamespace = config.ConsulNamespace
```

Comment:
The `ConsulNamespace` mapping code can probably happen as part of the validators if we do go ahead with that piece of the refactor later (I don't recall if it got mapped there or not in the initial refactor) -- but since it's a product of the gateway configuration (i.e. the namespace the gateway is created in) and some additional global configuration (i.e. the mapping configuration) -- it'd make sense to me to handle it as part of the initial "validation" step that calculates whatever overall state/status changes that will eventually get applied to the store/Consul/K8s statuses.